src/Storages/VirtualColumnUtils.cpp

```diff
@@ -115,7 +119,29 @@ NameSet getVirtualNamesForFileLikeStorage()
     return {"_path", "_file", "_size", "_time"};
 }
 
-VirtualColumnsDescription getVirtualsForFileLikeStorage(const ColumnsDescription & storage_columns)
+std::unordered_map<std::string, std::string> parseHivePartitioningKeysAndValues(const std::string& path, const ColumnsDescription & storage_columns)
```

Comment:
```suggestion
std::unordered_map<std::string, std::string> parseHivePartitioningKeysAndValues(const std::string & path, const ColumnsDescription & storage_columns)
```
Don't forget about the style

Reply:
Also, you can use `String` alias instead of `std::string`.